cmd/genji/shell/command.go

```diff
@@ -136,3 +142,193 @@ func displaySuggestions(in string) error {
 	fmt.Println()
 	return nil
 }
+
+// dumpTable displays the content of the given table as SQL statements.
+func dumpTable(tx *genji.Tx, t *database.Table) error {
+	var buf bytes.Buffer
+
+	c := fmt.Sprintf("CREATE TABLE %s", t.Name())
+	buf.WriteString(c)
+
+	ti, err := t.Info()
+	if err != nil {
+		return err
+	}
+
+	fcs := ti.FieldConstraints
+	hasField := false
+	for i, fc := range fcs {
+		// Fields constraints should be displaying between parenthesis.
+		if !hasField {
+			buf.WriteString("(\n")
+			hasField = true
+		}
+
+		// Don't put the last comma.
+		if i > 0 {
+			buf.WriteString(",\n")
+		}
+
+		buf.WriteString("  " + fcs[i].Path.String() + " ")
+		buf.WriteString(strings.ToUpper(fcs[i].Type.String()))
+		if fc.IsPrimaryKey {
+			buf.WriteString(" PRIMARY KEY")
+		}
+
+		if fc.IsNotNull {
+			buf.WriteString(" NOT NULL")
+		}
+	}
+
+	// Fields constraints close parenthesis.
+	if hasField {
+		buf.WriteString("\n);\n")
+	} else {
+		buf.WriteString(";\n")
+	}
+
+	// Print CREATE TABLE statement.
+	_, err = fmt.Fprintf(os.Stdout, buf.String())
+	if err != nil {
+		return err
+	}
+	buf.Reset()
+
+	// Indexes statement.
+	indexes, err := t.Indexes()
+	if err != nil {
+		return err
+	}
+
+	for _, index := range indexes {
+		u := ""
+		if index.Opts.Unique {
+			u = " UNIQUE"
+		}
+
+		info := fmt.Sprintf("CREATE%s INDEX %s ON %s (%s);\n", u, index.Opts.IndexName, index.Opts.TableName,
+			index.Opts.Path)
+		buf.WriteString(info)
+
+		_, err = fmt.Fprintf(os.Stdout, buf.String())
+		if err != nil {
+			return err
+		}
+		buf.Reset()
+	}
+
+	q := fmt.Sprintf("SELECT * FROM %s", t.Name())
+	res, err := tx.Query(q)
+	if err != nil {
+		return err
+	}
+	defer res.Close()
+
+	// Inserts statement.
+	insert := fmt.Sprintf("INSERT INTO %s VALUES ", t.Name())
+	return res.Iterate(func(d document.Document) error {
+		buf.WriteString(insert)
+
+		data, err := document.MarshalJSON(d)
+		if err != nil {
+			return err
+		}
+		buf.Write(data)
+		buf.WriteString(";\n")
+
+		if _, err = fmt.Fprintf(os.Stdout, buf.String()); err != nil {
+			return err
+		}
+
+		buf.Reset()
+
+		return nil
+	})
+}
+
+// runDumpCmd run .dump command where tables slice is the given tables or it can be empty that is consider all database.
+func runDumpCmd(db *genji.DB, tables []string) error {
+	tx, err := db.Begin(false)
+	if err != nil {
+		return err
+	}
+	defer tx.Rollback()
+
+	fmt.Println("BEGIN TRANSACTION;")
+
+	// Dump the given table(s) content.
+	argsTable := false
+	i := 0
+	for _, table := range tables {
+		argsTable = true
+		t, err := tx.GetTable(table)
+		switch err {
+		case nil:
+			// Blank separation between tables.
+			if i > 0 {
+				fmt.Println()
+			}
+			i++
+
+			if err := dumpTable(tx, t); err != nil {
+				fmt.Println("ROLLBACK;")
+				return err
+			}
+		case database.ErrTableNotFound: // If table doesn't exist we skip it.
+			break
```

Comment:
`break` is ambiguous because of its meaning in other languages, only use it when you want to explicitly stop the iteration on a loop. If you want to skip this loop iteration, use `continue` instead, or if you want to do nothing, just write nothing.

```suggestion
			continue
```

Reply:
You are right and `break` doesn't do what I want. Mistakenly wrote.